inventory-provisioning-core/src/main/scala/com/normation/inventory/ldap/provisioning/SoftwareIdFinder.scala

```diff
@@ -58,46 +58,44 @@ object NameAndVersionIdFinder {
  * Retrieve the id from the Mother Board Id
  */
 class NameAndVersionIdFinder(
-    ldapConnectionProvider:LDAPConnectionProvider[RoLDAPConnection],
-    mapper:InventoryMapper,
-    dit:InventoryDit
+    val name              : String
+  , ldapConnectionProvider: LDAPConnectionProvider[RoLDAPConnection]
+  , mapper                : InventoryMapper
+  , dit                   : InventoryDit
 ) extends SoftwareDNFinderAction {
 
   //the onlyTypes is an AND filter
-  override def tryWith(entity:Software) : Box[SoftwareUuid] = {
-    //create filter for software name and version
-    val nameFilter = entity.name match {
-      case None => NOT(HAS(A_NAME))
-      case Some(x) => EQ(A_NAME,x)
-    }
+  override def tryWith(entities:Set[Software]) : Box[MergedSoftware] = {
 
-    val versionFilter = entity.version match {
-      case None => NOT(HAS(A_SOFT_VERSION))
-      case Some(x) => EQ(A_SOFT_VERSION,x.value)
-    }
 
-    val filter = AND(nameFilter,versionFilter)
+    val filter = OR(entities.map { entity =>
+      //create filter for software name and version
+      val nameFilter = entity.name match {
+        case None => NOT(HAS(A_NAME))
+        case Some(x) => EQ(A_NAME,x)
+      }
+
+      val versionFilter = entity.version match {
+        case None => NOT(HAS(A_SOFT_VERSION))
+        case Some(x) => EQ(A_SOFT_VERSION,x.value)
+      }
```

Comment:
No clear, it depends on implementation but it's big (tens of megabytes ?)
See for ex: http://stackoverflow.com/questions/553293/what-is-the-ldap-filter-string-length-limit-in-active-directory or http://www.openldap.org/lists/openldap-technical/201008/msg00137.html

So it should not be a problems, even with thousands of software (I tested with inventories with at least 800 software)